nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/analytics/StatusAnalyticsModelMapFactory.java

```diff
@@ -0,0 +1,126 @@
+    private final static String QUEUED_COUNT_METRIC = "queuedCount";
+    private final static String QUEUED_BYTES_METRIC = "queuedBytes";
+    private final static String INPUT_COUNT_METRIC = "inputCount";
+    private final static String INPUT_BYTES_METRIC = "inputBytes";
+    private final static String OUTPUT_COUNT_METRIC = "outputCount";
+    private final static String OUTPUT_BYTES_METRIC = "outputBytes";
+
+    /**
+     * Return mapping of models and extraction functions for connection status analytics prediction instances
+     * @param extensionManager Extension Manager object for instantiating classes
+     * @param niFiProperties NiFi Properties object
+     * @return
+     */
+    public static Map<String, Tuple<StatusAnalyticsModel, StatusMetricExtractFunction>> getConnectionStatusModelMap(ExtensionManager extensionManager, NiFiProperties niFiProperties){
+            Map<String, Tuple<StatusAnalyticsModel, StatusMetricExtractFunction>> modelMap = new HashMap<>();
+            StatusMetricExtractFunction extract = getConnectionStatusExtractFunction();
+            Tuple<StatusAnalyticsModel, StatusMetricExtractFunction> countModelFunction = new Tuple<>(createModelInstance(extensionManager, niFiProperties), extract);
+            Tuple<StatusAnalyticsModel, StatusMetricExtractFunction> byteModelFunction = new Tuple<>(createModelInstance(extensionManager, niFiProperties), extract);
+            modelMap.put(QUEUED_COUNT_METRIC, countModelFunction);
+            modelMap.put(QUEUED_BYTES_METRIC, byteModelFunction);
+            return modelMap;
+    }
+
+    /**
+     * Create a connection model instance  using configurations set in NiFi properties
+     * @param extensionManager Extension Manager object for instantiating classes
+     * @param nifiProperties NiFi Properties object
+     * @return statusAnalyticsModel
+     */
+    private static StatusAnalyticsModel createModelInstance(ExtensionManager extensionManager, NiFiProperties nifiProperties) {
+        final String implementationClassName = nifiProperties.getProperty(NiFiProperties.ANALYTICS_CONNECTION_MODEL_IMPLEMENTATION, NiFiProperties.DEFAULT_ANALYTICS_CONNECTION_MODEL_IMPLEMENTATION);
+        if (implementationClassName == null) {
+            throw new RuntimeException("Cannot create Analytics Model because the NiFi Properties is missing the following property: "
+                    + NiFiProperties.ANALYTICS_CONNECTION_MODEL_IMPLEMENTATION);
+        }
+        try {
+            return NarThreadContextClassLoader.createInstance(extensionManager, implementationClassName, StatusAnalyticsModel.class, nifiProperties);
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Get a connection status extract function instance
+     * @return StatusMetricExtractFunction
+     */
+    private static StatusMetricExtractFunction getConnectionStatusExtractFunction() {
+
+        return (metric, statusHistory) -> {
```

Comment:
Again, we should avoid the use of the Functional-style lambda when it doesn't save us much. Google's documentation of Guava does a great job of explaining the tradeoffs to consider when contemplating whether Functional or Imperative style should be used: https://github.com/google/guava/wiki/FunctionalExplained